code/modules/client/client_procs.dm

```diff
@@ -185,9 +185,6 @@ var/global/list/localhost_addresses = list(
 	prefs = SScharacter_setup.preferences_datums[ckey]
 	if(!prefs)
 		prefs = new /datum/preferences(src)
-	prefs.last_ip = address				//these are gonna be used for banning
-	prefs.last_id = computer_id			//these are gonna be used for banning
-	apply_fps(prefs.clientfps)
```

Comment:
Where is this stuff called if `prefs` exists (i.e. if line 187 isn't reached)?

Reply:
Moved to deferred_login()

Reply:
That doesn't work: the ip logging needs to happen immediately, it should not be deferrred.

Reply:
Moved back.